lib/honeybadger/logger.ex

```diff
@@ -86,6 +87,13 @@ defmodule Honeybadger.Logger do
 
   def domain_ignored?(_domain, _ignored), do: false
 
+  def internal_error?(metadata) do
+    case Keyword.get(metadata, :application) do
+      nil -> false
+      application -> application == :honeybadger
+    end
+  end
```

Comment:
Could we eliminate the conditional by using pattern matching here?

```suggestion
def internal_error?(%{application: :honeybadger}), do: true
def internal_error?(_metadata), do: false
```

Reply:
Working on it